packages/opentelemetry-core/test/internal/validators.test.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as assert from 'assert';
+import { validateKey, validateValue } from '../../src/internal/validators';
+
+describe('validators', () => {
+  describe('validateKey', () => {
+    it('returns true when key contain all allowed characters', () => {
+      const allowedKeys = 'abcdefghijklmnopqrstuvwxyz0123456789-_*/';
+      assert.ok(validateKey(allowedKeys));
+    });
+
+    it('returns false when invalid first key character', () => {
+      assert.ok(!validateKey('1_key'));
+    });
+
+    it('returns false when invalid key characters', () => {
+      assert.ok(!validateKey('kEy_1'));
+    });
+
+    it('returns false when invalid key size', () => {
+      assert.ok(!validateKey('k'.repeat(257)));
+    });
+  });
+
+  describe('validateValue', () => {
```

Comment:
Could you add a couple valid condition checks that include characters at the borders of the allowed range as well as in the middle of it (and including a space in the middle)? Would also be nice to see characters at the beginning and end of range at the end since that's a separate part of the regex. I think all these valid ones could be in one `it` block.

Could you also add conditions that it can't contain an equals or comma at the end?